### DISCUSSION TITLE
How to bundle patched kubelet?

### DISCUSSION THREAD

**Discussion by techthumb** [bottlerocket-os/bottlerocket#4119] [Q&A] [answered]
Hi Folks,

Many thanks for open sourcing this project.

We have a requirement where we want to patch kubernetes/kubelet. We have:
1. placed the patch in https://github.com/bottlerocket-os/bottlerocket-core-kit/tree/develop/packages/kubernetes-1.30
2. added it to https://github.com/bottlerocket-os/bottlerocket-core-kit/blob/develop/packages/kubernetes-1.30/kubernetes-1.30.spec

`make` succeeds and the .rpm produced has the patched version of the kubelet.

In https://github.com/bottlerocket-os/bottlerocket, `cargo make ami` successfully publishes the AMI.

Question: How do we make the `cargo make ami` step aware of the custom RPM?

TIA

**Comment by larvacea** [comment, marked as answer]
Subject to change, since this process is new and possibly evolving, but here are the missing pieces:

## Core kit build and publish
* Create a file `Infra.toml` in the root of your core kit working tree. This file is in the `.gitignore` list. The file should contain:
```
[vendor.bottlerocket]
registry = "something.dkr.ecr.us-west-2.amazonaws.com"
```
* As you already found, `make ARCH=aarch64 && make ARCH=x86_64` will build the kit. You may build either or both architectures to suit your needs.
* `make publish` will tag and upload your custom kit to the registry `something.dkr.ecr.us-west-2.amazonaws.com` where you will have to replace that registry with something more suitable.

## AMI build, consuming custom kit.
* Edit the `Twoliter.toml` file in your `bottlerocket` working tree to include:
```
[vendor.yourname]
registry = "something.dkr.ecr.us-west-2.amazonaws.com"
```
and edit the piece starting with `[[kit]]` to match:
```
[[kit]]
name = "bottlerocket-core-kit"
version = "2.3.0"
vendor = "yourname"
```
* After this edit, the `cargo make` command in this `bottlerocket` working tree will consume your custom kit, and then `cargo make ami` can be used to publish it to EC2 as usual.

Please do let us know how this works for you.

**Reply by techthumb** [reply]
Worked flawlessly on `develop`. Many thanks for the prompt & accurate information.
